kratos/linear_solvers/fallback_linear_solver.h

```diff
@@ -0,0 +1,897 @@
+    }
+
+    /**
+     * @brief Get the default parameters for this solver.
+     * @details This function returns the default parameters for configuring this solver.
+     * @return Default parameters for the solver.
+     */
+    const Parameters GetDefaultParameters() const
+    {
+        return Parameters(R"({
+            "solver_type": "fallback_linear_solver",
+            "solvers"    : [
+                // Empty in defaults. Should be filled with the solvers to try. For example:
+                // {
+                //     "solver_type": "amgcl"
+                // },
+                // {
+                //     "solver_type": "skyline_lu_factorization"
+                // }
+                // Label of the solver is solver_x, where x is the index in the list
+            ],
+            "reset_solver_index_each_try": false
+        })");
+    }
+
+    /**
+     * @brief Updates the solver index counter and logs transitions between solvers.
+     * @details This method increments the current solver index to switch to the next solver in a sequence of fallback solvers.
+     * It logs both the settings of the solver that failed (if any) and the settings of the next solver to be used.
+     * The method is designed to facilitate easy tracking and debugging of solver transitions within a sequence.
+     * @note This method now includes safety checks to prevent out-of-bounds access to the solver array. It also
+     *       outlines a placeholder for enhanced future functionality, such as more comprehensive logging or
+     *       additional transition actions.
+     */
+    void UpdateCounterSolverIndex()
+    {
+        // Safety check to ensure we have solvers to work with
+        if (mSolvers.empty()) {
+            KRATOS_WARNING("FallbackLinearSolver") << "No solvers are configured." << std::endl;
+            return;
+        }
+
+        // Log the settings of the current (failing) solver, if applicable
+        if (mCurrentSolverIndex < mSolvers.size()) {
+            KRATOS_INFO("FallbackLinearSolver") << "Current solver " << GetCurrentSolver()->Info() << " failed with the following settings: " << mParameters["solvers"][mCurrentSolverIndex].PrettyPrintJsonString() << std::endl;
+        } else {
+            KRATOS_WARNING("FallbackLinearSolver") << "Current solver index is out of bounds." << std::endl;
+            return;
+        }
+
+        // Increment the counter
+        mCurrentSolverIndex++;
+
+        // Print the new solver settings
+        if (mCurrentSolverIndex < mSolvers.size()) {
+            KRATOS_INFO("FallbackLinearSolver") << "Switching to new solver " << GetCurrentSolver()->Info() << " with the following settings: " << mParameters["solvers"][mCurrentSolverIndex].PrettyPrintJsonString() << std::endl;
+        } else {
+            KRATOS_WARNING("FallbackLinearSolver") << "New solver index is out of bounds." << std::endl;
```

Comment:
again, just throw an error

Reply:
Nope, with this it will simply continues as it is done now with current standard linear solvers

Reply:
I would say that this method should not even be called when the last solver is reached. Seems too complicated, and you need to handle the out-of-bounds case in other places too

Reply:
I agree with @loumalouomega 's observation. If the last one (i.e., the more robust one) is reached and convergence is not achieved you stick to that and you go on, something equivalent to current behavior.

Nevertheless, I also agree with @philbucher 's observation. Theoretically, we shouldn't call this if we are in the last one. In any case, this is in my opinion a tiny detail in the sense that you either handle `mCurrentSolverIndex < mSolvers.size()` inside or outside this function. Hence, this is not a blocker for me.

However, I'd like to note that either you handle `mCurrentSolverIndex < mSolvers.size()` inside or outside the warning message `New solver index is out of bounds.` is not very descriptive, or even misleading (there's new solver index but we keep the old one). I'd modify the message to be more self-descriptive of what's actually happening.